spanner/google/cloud/spanner_v1/keyset.py

```diff
@@ -60,17 +60,19 @@ def to_pb(self):
         """
         kwargs = {}
 
-        if self.start_open:
+        if self.start_open is not None:
             kwargs['start_open'] = _make_list_value_pb(self.start_open)
-
-        if self.start_closed:
+        elif self.start_closed is not None:
```

Comment:
I don't think `elif` is what you want here?

Reply:
it's been so long, but I think elif is right.  I think I was guarding against the case for [] where neither self.start_open and self.start_closed would not be hit, and therefore passing to pb will complain that there was a missing key.

Reply:
I don't think it matters in this case whether it's an if or an elif (since ValueError gets raised earlier), but elif makes it clear that it should either be a start_open, or a start_closed.

Reply:
There is a slight difference when using an explicitly empty **message field**:

```python
>>> from google.protobuf.struct_pb2 import ListValue
>>> from google.cloud.spanner_v1.proto.keys_pb2 import KeyRange
>>> k1 = KeyRange()
>>> k2 = KeyRange(
...     start_closed=ListValue(values=[]),
...     end_closed=ListValue(values=[]),
... )
>>> k1

>>> k2
start_closed {
}
end_closed {
}

>>>
```

As opposed to a simple field:

```python
>>> from google.cloud.spanner_v1.proto.keys_pb2 import KeySet
>>> k1 = KeySet()
>>> k2 = KeySet(all=False)
>>> k3 = KeySet(all=True)
>>> k1

>>> k2

>>> k3
all: true

>>>
```